sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -272,8 +260,7 @@ static bool isDeviceBinaryTypeSupported(const context &C,
   }
 
   // OpenCL 2.1 and greater require clCreateProgramWithIL
-  backend CBackend = (detail::getSyclObjImpl(C)->getPlugin()).getBackend();
-  if ((CBackend == backend::opencl) &&
+  if ((ContextBackend == backend::opencl) &&
       C.get_platform().get_info<info::platform::version>() >= "2.1")
```

Comment:
Just curious does this really work as it's intended to?
Will `operator>=` for `std::string` be called here?